lib/Doctrine/DBAL/Types/DecimalType.php

```diff
@@ -36,7 +37,7 @@ public function convertToPHPValue($value, AbstractPlatform $platform)
     {
         // Some drivers starting from PHP 8.1 can represent decimals as float
         // See also: https://github.com/doctrine/dbal/pull/4818
-        if (PHP_VERSION_ID >= 80100 && is_float($value)) {
+        if (PHP_VERSION_ID >= 80100 && (is_float($value) || is_int($value))) {
```

Comment:
Do we really need to enumerate all types that need to be converted to a string? Would it make sense to replace it with something like `$value !== null && !is_string($value)`?

Reply:
I'd prefer to be caucious here. Integer and float are really the only non-string types that we expect here.